semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/redis/RedisHashSetVectorStoreRecordCollectionOptions.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    /**
+     * Gets the vector store record mapper.
+     *
+     * @return the vector store record mapper
+     */
+    @Nullable
+    public VectorStoreRecordMapper<Record, Entry<String, Map<String, String>>> getVectorStoreRecordMapper() {
+        return vectorStoreRecordMapper;
+    }
+
+    /**
+     * Gets whether to prefix the collection name to the redis key.
+     *
+     * @return whether to prefix the collection name to the redis key
+     */
+    public boolean isPrefixCollectionName() {
+        return prefixCollectionName;
+    }
+
+    /**
+     * Builder for {@link RedisHashSetVectorStoreRecordCollectionOptions}.
+     *
+     * @param <Record> the record type
+     */
+    public static class Builder<Record> {
+        @Nullable
+        private VectorStoreRecordMapper<Record, Entry<String, Map<String, String>>> vectorStoreRecordMapper;
+        @Nullable
+        private Class<Record> recordClass;
+        @Nullable
+        private VectorStoreRecordDefinition recordDefinition;
+        private boolean prefixCollectionName = true;
+
+        /**
+         * Sets the record class.
+         *
+         * @param recordClass the record class
+         * @return the builder
+         */
+        public Builder<Record> withRecordClass(Class<Record> recordClass) {
+            this.recordClass = recordClass;
+            return this;
+        }
+
+        /**
+         * Sets the vector store record mapper.
+         *
+         * @param vectorStoreRecordMapper the vector store record mapper
+         * @return the builder
+         */
+        public Builder<Record> withVectorStoreRecordMapper(
+            VectorStoreRecordMapper<Record, Entry<String, Map<String, String>>> vectorStoreRecordMapper) {
+            this.vectorStoreRecordMapper = vectorStoreRecordMapper;
+            return this;
+        }
+
+        /**
+         * Sets the record definition.
+         *
+         * @param recordDefinition the record definition
+         * @return the builder
+         */
+        public Builder<Record> withRecordDefinition(VectorStoreRecordDefinition recordDefinition) {
+            this.recordDefinition = recordDefinition;
+            return this;
+        }
+
+        /**
+         * Sets whether to prefix the collection name to the redis key.
+         * Default is true.
+         *
+         * @param prefixCollectionName whether to prefix the collection name to the redis key
+         * @return the builder
+         */
+        public Builder<Record> withPrefixCollectionName(boolean prefixCollectionName) {
+            this.prefixCollectionName = prefixCollectionName;
+            return this;
+        }
+
+        /**
+         * Builds the options.
+         *
+         * @return the options
+         */
+        public RedisHashSetVectorStoreRecordCollectionOptions<Record> build() {
+            if (recordClass == null) {
+                throw new IllegalArgumentException("recordClass must be provided");
```

Comment:
Make an sk exception

Reply:
This will be added in a separate PR https://github.com/microsoft/semantic-kernel-java/issues/194